scipy/optimize/_numdiff.py

```diff
@@ -240,17 +252,24 @@ def approx_derivative(fun, x0, method='3-point', rel_step=None, f0=None,
 
         Note, that sparse differencing makes sense only for large Jacobian
         matrices where each row contains few non-zero elements.
+    return_linear_operator : boolean, optional
+        When `return_linear_operator` is True returns an LinearOperator.
+        Otherwise it returns an dense array or sparse matrix depending
+        on how `sparsity` is defined. By default is False.
     args, kwargs : tuple and dict, optional
         Additional arguments passed to `fun`. Both empty by default.
         The calling signature is ``fun(x, *args, **kwargs)``.
 
     Returns
     -------
-    J : ndarray or csr_matrix
-        Finite difference approximation of the Jacobian matrix. If `sparsity`
-        is None then ndarray with shape (m, n) is returned. Although if m=1 it
-        is returned as a gradient with shape (n,). If `sparsity` is not None,
-        csr_matrix with shape (m, n) is returned.
+    J : {ndarray, sparse matrix, LinearOperator}
+        Finite difference approximation of the Jacobian matrix.
```

Comment:
Some clarification here is also useful. 

For example, what if we have a scalar function, then linear operator will return scalars as well (0-d arrays)?